# Type: feat
# Subtype: E1
# English Summary: If orderSetting is not empty, set order value to orderSetting
# Chinese Summary: 如果orderSetting不为空，则设置order值为orderSetting
# Entry Number: 21

# === Git Diff Content ===

diff --git a/grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/DefaultGrailsCodecClass.java b/grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/DefaultGrailsCodecClass.java
index 50ead26109..fd7b042508 100644
--- a/grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/DefaultGrailsCodecClass.java
+++ b/grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/DefaultGrailsCodecClass.java
@@ -64,6 +64,9 @@ public class DefaultGrailsCodecClass extends AbstractInjectableGrailsClass imple
 
     private void initializeCodec() {
         Integer orderSetting = (Integer)getPropertyOrStaticPropertyOrFieldValue("order", Integer.class);
+        if(orderSetting != null) {
+            order = orderSetting;
+        }
         if(Encoder.class.isAssignableFrom(getClazz())) {
             encoder = (Encoder)getReferenceInstance();
             autowireCodecBean(encoder);
